Add a drake_vendor namespace ala tools/workspace/vendor_cxx

This prevents link-time symbol conflicts in case code downstream of
Drake wants to use a different build of nanoflann.

--- include/nanoflann.hpp.orig
+++ include/nanoflann.hpp
@@ -70,6 +70,7 @@
 #endif
 #endif
 
+inline namespace drake_vendor __attribute__ ((visibility ("hidden"))) {
 namespace nanoflann
 {
 /** @addtogroup nanoflann_grp nanoflann C++ library for ANN
@@ -2414,3 +2415,4 @@
 
 /** @} */  // end of grouping
 }  // namespace nanoflann
+}  // namespace drake_vendor
\ No newline at end of file
